specs/agents/span-links.md

```diff
@@ -0,0 +1,36 @@
+## Span Links
+
+A Span or Transaction MAY link to zero or more other Spans/Transactions that are causally related.
+
+Example use-cases for Span Links:
+
+1. When a single transaction represents the batch processing of several messages, the agent SHOULD be able to link back to the traces that have produced the messages.
+2. When the agent receives a `traceparent` header from outside a trust boundary, it SHOULD restart the trace (creating a different trace id with its own sampling decision) and link to the originating trace.
+3. Close gap for the OTLP intake - [OTel's specification of span links](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#links-between-spans)
+
+Spans and Transactions MUST collect links in the `links` array with the following fields on each item:
```

Comment:
I wonder if the link should have type/kind to better indicate why a trace/span is linked to the current one.

Reply:
The labels field discussed above could potentially cover that: https://github.com/elastic/apm/pull/600#pullrequestreview-889811169